core/types/transaction.go

```diff
@@ -610,6 +639,21 @@ func (t *TransactionsByPriceAndNonce) Shift() {
 	heap.Pop(&t.heads)
 }
 
+// ShiftNoSorting replaces the current best head with the next one from the same account without sorting by fee.
+func (t *TransactionsByPriceAndNonce) ShiftNoSorting(acc common.AddressBytes) {
```

Comment:
Same comment as above, I would prefer to just add a `sort` flag to `Shift(sort bool)` to optionally sort